packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -114,9 +110,50 @@ class TimeOfDay {
   @override
   int get hashCode => hashValues(hour, minute);
 
-  // TODO(ianh): Localize.
   @override
-  String toString() => '$hourOfPeriodLabel:$minuteLabel $periodLabel';
+  String toString() => '$hourLabel:$minuteLabel';
+}
+
+/// The list of time of day formats supported by this library encoded as ICU
+/// "Short Time" patterns.
+///
+/// The time picker invoked via [showTimePicker] has configurations optimized
+/// for these formats. Any other formats will default to [h_colon_mm_space_a].
+/// The time picker get the value from [MaterialLocalizations.timeOfDayFormat].
+abstract class TimeOfDayFormats {
+  // Patterns using HH.
+
+  /// One of the reusable time patterns and also the default format.
+  ///
+  /// This format is used as a fallback when the value of
+  /// [MaterialLocalizations.timeOfDayFormat] is not supported, i.e. not one of
+  /// the patterns listed in this class.
+  static const String HH_colon_mm = 'HH:mm';
+
+  /// The 'HH.mm' pattern.
+  static const String HH_dot_mm = 'HH.mm';
```

Comment:
These docs don't comply to our style guide (you could write them without knowing anything but the name of the member).

Maybe this whole class should just be private?

Reply:
Made patterns private, but created an enum of supported layouts.